App/Redux/PreferencesRedux.ts

```diff
@@ -24,20 +25,28 @@ const actions = {
     return (publicKey: string) => resolve({ publicKey })
   }),
   completeTourSuccess: createAction('COMPLETE_TOUR_SUCCESS', resolve => {
-  return (tourKey: string) => resolve({ tourKey })
-}),
+    return (tourKey: TourScreens) => resolve({ tourKey })
+  }),
+  toggleServicesRequest: createAction('TOGGLE_SERVICES_REQUEST', resolve => {
+    return (name: ServiceType, status?: boolean) => resolve({ name, status })
+  })
 }
 
 export type PreferencesAction = ActionType<typeof actions>
 
+export type TourScreens = 'wallet' | 'threads' | 'notifications' | 'feed'
+export type ServiceType = 'backgroundLocation' | 'notifications' | 'photoAddedNotification' | 'receivedInviteNotification' | 'deviceAddedNotification' | 'commentAddedNotification' | 'likeAddedNotification' | 'peerJoinedNotification' | 'peerLeftNotification'
+export type Service = {
+  status: boolean,
+}
 export type PreferencesState = {
   onboarded: boolean
   verboseUi: boolean
   mnemonic?: string
   publicKey?: string
   profile?: Profile
   pending?: string,
-  // tracks if a user has seen a particular tour screen
+  readonly services: {[index: string]: Service}
```

Comment:
Maybe this should be indexed by `ServiceType` and not `string`?

Reply:
you cannot have both in the map like that

Reply:
as in, `{[index: ServiceType]: Service}` isn't valid

Reply:
Hmm. Maybe if you make the enum?

Reply:
could be yeah, sounds like future optimization. by using the `ServiceType` for all action inputs where it occurs it's 99% typed :)

Reply:
Sorry, i meant maybe if you make `ServiceType` an `enum`.

Reply:
ah yeah, could be

Reply:
just tried, must be string or number

Reply:
make all of services an enum i guess. seriously though, next time

Reply:
ha